src/components/Providers.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { MovieProviders } from '../types/tmdb';
+import { getMovieProviders } from '../helpers/getMovieUtils';
+
+interface ProviderProps { ID: number | null }
```

Comment:
```suggestion
interface ProviderProps { ID: number }
```
If we make the ID prop required, we can avoid ever calling the component improperly. This **should** make it more resilient to issues. We would need to update the invocation of `<Providers>` to be:
```ts
{ details.id && <Providers id={details.id} /> }
```